python/ray/serve/controller.py

```diff
@@ -555,15 +572,100 @@ def get_app_config(self) -> Dict:
                 "deployments": [],
             }
         else:
-            _, config = pickle.loads(checkpoint)
+            _, config, _ = pickle.loads(checkpoint)
             return config
 
 
+def _version_config(
```

Comment:
The comments looks good. Let's now change the function name so it's more readable. How about 
```suggestion
def _generate_new_version_config(
```